src/main/kotlin/net/irisshaders/lilybot/extensions/util/ThreadInviter.kt

```diff
@@ -117,29 +115,22 @@ class ThreadInviter : Extension() {
 			check { failIf(event.channel.member != null) } // To avoid running on thread join, rather than creation only
 
 			action {
-				var supportError = false
-				var moderatorRoleError = false
+				// Try to get the moderator ping role from the config. If a config is not set, return@action
+				val moderatorRoleId = DatabaseHelper.selectInConfig(event.channel.guildId.toString(), "moderatorsPing")
 
-				val supportChannel = DatabaseHelper.selectInConfig(event.channel.guild.id, DatabaseManager.Config.supportChannel)
-				val supportTeamId = DatabaseHelper.selectInConfig(event.channel.guild.id, DatabaseManager.Config.supportTeam)
-				val moderatorRole = DatabaseHelper.selectInConfig(event.channel.guild.id, DatabaseManager.Config.moderatorsPing)
-
-				if (supportChannel.equals("NoSuchElementException") || supportTeamId.equals("NoSuchElementException")) {
-					supportError = true
-				} else if (moderatorRole.equals("NoSuchElementException")) {
-					moderatorRoleError = true
-				}
+				val supportTeamId = DatabaseHelper.selectInConfig(event.channel.guildId.toString(), "supportTeam")
+				val supportChannelId = DatabaseHelper.selectInConfig(event.channel.guildId.toString(), "supportChannel")
 
-				if (!supportError) {
+				if (supportTeamId != null && supportChannelId != null) {
```

Comment:
Thinking about it, this should probably be an or, to catch incase anybody sets one or the other

Reply:
It's checking if they're both not null, i.e. both have been set. An or would mean that one could possibly be null, and that wouldn't work.